src/types.rs

```diff
@@ -655,6 +664,14 @@ impl SourceMap {
         }
     }
 
+    pub fn add_to_ignore_list(&mut self, src_id: u32) {
+        self.ignore_list.insert(src_id);
+    }
+
+    pub fn ignore_list(&self) -> btree_set::Iter<'_, u32> {
```

Comment:
```suggestion
    pub fn ignore_list(&self) -> impl Iterator<Item=u32> + '_ {
```

This hides the implementation details, and makes it possible to change.